GroceryStore/GroceryStore.Models/Customer.cs

```diff
@@ -0,0 +1,113 @@
+﻿using GroceryStore.Constants;
+using System.Diagnostics;
+using System.Dynamic;
+using System.Globalization;
+using System.Transactions;
+using System.Xml.Linq;
+
+namespace GroceryStore.Models
+{
+    public class Customer
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public int Age { get; set; }
+        public string Sex { get; set; }
+
+        private bool _hasDiscountCard;
+        public bool HasDiscountCard
+        {
+            get
+            { return _hasDiscountCard; }
+            set
+            { _hasDiscountCard = value; }
+        }
+        public string DiscountCardToString
+        {
+            get
+            {
+                return _hasDiscountCard ? "Yes" : "No";
+            }
+        }
+        private double _personalDiscount;
+        public double PersonalDiscount
+        {
+            get
+            {
+                return (HasDiscountCard ? _personalDiscount : 0);
+            }
+            set
+            {
+                _personalDiscount = value;
+            }
+        }
+        public string PercentageAsString
+        {
+            get
+            {
+                return (PersonalDiscount * 100).ToString("0.##") + "%";
+            }
+        }
+        public string FullName
+        {
+            get
+            {
+                return $"{FirstName} {LastName}";
+            }
+        }
+        public Product[] Cart { get; set; }
+        public int CartCount { get; set; }
+
+
+        public Customer(string firstName, string lastName, int age, string sex, bool hasDiscountCard, double personalDiscount)
```

Comment:
You need to add some default value (e.g 0.02) for personalDiscount parameter

[Named and Optional arguments article](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)